src/storage/src/hummock/store/state_store.rs

```diff
@@ -186,14 +187,27 @@ impl StateStoreWrite for LocalHummockStorage {
             let epoch = write_options.epoch;
             let table_id = write_options.table_id;
 
+            let sorted_items = SharedBufferBatch::build_shared_buffer_item_batches(kv_pairs);
+            let size = SharedBufferBatch::measure_batch_size(&sorted_items);
+            let limiter = self.core.memory_limiter.as_ref();
+            let tracker = if let Some(tracker) = limiter.try_require_memory(size as u64) {
+                tracker
+            } else {
+                self.core
+                    .event_sender
+                    .send(HummockEvent::BufferMayFlush)
```

Comment:
Also, do we also need to send `BufferMayFlush` when we successfully aquire the memory and the memory usage is above flush threshold?

Reply:
Yes, seems that we miss sending the Flush event before. 

And I think we can add a check on flush threshold in the event handler every time we receive an imm.

Reply:
I just check the code in release 0.1.14, which uses the local version manager to handle the upload. It seems that after there is data added to local version manager, it will send a Flush event. So we only need to add this logic of checking threshold in the main branch.